inspirehep/modules/orcid/schema.py

```diff
@@ -0,0 +1,126 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+import dojson
+
+from inspirehep.utils.record import get_abstract, get_subtitle, get_title
+
+orcid_overdo = dojson.Overdo()
+
+
+@orcid_overdo.over('title', 'titles')
+def title_rule(self, key, value):
+    return {"title": get_title({value}),
+            "subtitle": get_subtitle({value})}
+
+
+@orcid_overdo.over('journal_title', 'publication_info')
+def publication_rule(self, key, value):
+    try:
+        return value[0].get('journal_title')
+    except (TypeError, KeyError):
+        pass
+
+
+@orcid_overdo.over('short_description', 'abstracts')
+def abstract_rule(self, key, value):
+    try:
+        return get_abstract({value})
+    except (TypeError, KeyError):
+        pass
+
+
+@orcid_overdo.over('publication_date', 'imprints')
+def date_rule(self, key, value):
+    try:
+        date = value[0].get('date')
+        date = date.split('-')
+        final_date = ['', '', '']
+        if date[0]:
+            final_date[0] = date[0]
+        if date[1]:
+            final_date[1] = date[1]
+        if date[2]:
+            final_date[2] = date[2]
+        publication_date = {'year': int(final_date[0]),
+                            'month': int(final_date[1]),
+                            'day': int(final_date[2])
+                            }
+        return publication_date
+    except (TypeError, IndexError):
+        pass
+
+
+@orcid_overdo.over('work_external_identifier', 'arxiv_eprints|dois')
+@dojson.utils.for_each_value
+def external_id_rule(self, key, value):
+    if key == 'dois':
+        return{
+            'external-identifier-type': 'DOI',
+            'external-identifier-id': value.get('value')
+        }
+    if key == 'arxiv_eprints':
+        return {
+            'external-identifier-type': 'ARXIV',
+            'external-identifier-id': value.get('value')
+        }
+
+
+@orcid_overdo.over('contributor', 'authors')
+@dojson.utils.for_each_value
+def authors_rule(self, key, value):
+    try:
+        return {
+            'credit-name': value.get('full_name'),
+            'contributor-orcid': value.get('orcid') if 'orcid' in value else '',
+            'contributor-attributes': {
+                'contributor-role': 'AUTHOR',
+                # One of 'ADDITIONAL', 'FIRST'
+                'contributor-sequence': 'ADDITIONAL'
```

Comment:
@kaplun Regarding this what should we do? `FIRST` means main authors and `ADDITIONAL` means co-authors, as far as I understood. Should we mark every author as `ADDITIONAL` or  should we not provide the information?